frontend/src/helpers/user/user_api_wrappers.tsx

```diff
@@ -1,25 +1,295 @@
+    "createUser",
+    response.message,
+    PeerPrepErrors.InternalServerError
+  );
+};
+
+const updateUser = async (id: string, user: User) => {
+  // call PUT /api/users/:id from user service
+  const response = await api({
+    method: HTTP_METHODS.PUT,
+    service: service,
+    path: id,
+    body: user,
+    tags: scope,
+  });
+
+  // successful response should return 204
+  if (response.status === HttpStatusCode.NO_CONTENT) {
+    // revalidateTag(SERVICE.USER);
+    return true;
+  } else if (response.status === HttpStatusCode.BAD_REQUEST) {
+    return throwAndLogError(
+      "updateUser",
+      response.message,
+      PeerPrepErrors.BadRequestError
+    );
+  } else if (response.status === HttpStatusCode.NOT_FOUND) {
+    return throwAndLogError(
+      "updateUser",
+      response.message,
+      PeerPrepErrors.NotFoundError
+    );
+  } else if (response.status === HttpStatusCode.CONFLICT) {
+    return throwAndLogError(
+      "updateUser",
+      response.message,
+      PeerPrepErrors.ConflictError
+    );
+  }
+  return throwAndLogError(
+    "updateUser",
+    response.message,
+    PeerPrepErrors.InternalServerError
+  );
+};
+
+const deleteUser = async (id: string) => {
+  // call DELETE /api/users/:id from user service
+  const response = await api({
+    method: HTTP_METHODS.DELETE,
+    service: service,
+    path: id,
+    tags: scope,
+  });
+
+  // successful response should return 204
+  if (response.status === HttpStatusCode.NO_CONTENT) {
+    return true;
+  } else if (response.status === HttpStatusCode.NOT_FOUND) {
+    return throwAndLogError(
+      "deleteUser",
+      response.message,
+      PeerPrepErrors.NotFoundError
+    );
+  }
+  return throwAndLogError(
+    "deleteUser",
+    response.message,
+    PeerPrepErrors.InternalServerError
+  );
 };
 
-const getUsername = () => {
-  return "test user";
+const getUserPreferenceById = async (id: string) => {
+  // call GET /api/users/:id/preferences from user service
+  const response = await api({
+    method: HTTP_METHODS.GET,
+    service: service,
+    path: `${id}/preferences`,
+    tags: scope,
+  });
+
+  // successful response should return 200 with the user preference data
+  if (response.status === HttpStatusCode.OK) {
+    const userPreference = response.data as Preference;
+    logger.info(`[getUserPreferenceById(${id})] ${userPreference}`);
+    return userPreference;
+  } else if (response.status === HttpStatusCode.NOT_FOUND) {
+    return throwAndLogError(
+      "getUserPreferenceById",
+      response.message,
+      PeerPrepErrors.NotFoundError
+    );
+  }
+  return throwAndLogError(
+    "getUserPreferenceById",
+    response.message,
+    PeerPrepErrors.InternalServerError
+  );
 };
 
-const getUserPreferences = () => {
-  return {
-    languages: ["Python", "C++"],
-    difficulties: ["Easy", "Medium"],
-    topics: ["Array", "String", "Tree"],
-  };
+const createUserPreference = async (id: string, userPreference: Preference) => {
```

Comment:
This `createUserPreference` will no longer be used since I had removed the `POST /users/:userId/preferences` endpoint. Thus we can just remove this API wrapper fyi.